corehq/ex-submodules/casexml/apps/phone/utils.py

```diff
@@ -51,11 +52,15 @@ def get_or_cache_global_fixture(restore_state, cache_bucket_prefix, fixture_name
         data = get_cached_fixture_items(domain, cache_bucket_prefix)
 
     if data is None:
-        items = data_fn()
-        io_data = write_fixture_items_to_io(items)
-        data = io_data.read()
-        io_data.seek(0)
-        cache_fixture_items_data(io_data, domain, fixture_name, cache_bucket_prefix)
+        with CriticalSection('{}/{}'.format(cache_bucket_prefix, domain), block=False) as lock:
+            if not lock.success():
+                return []  # if we can't get the lock don't include this fixture in the restore
```

Comment:
would this mean that a user will not get a fixture in the restore if its being processed by another restore request?
If yes, then the restore would fail or it would succeed but the app would not have it?

Reply:
it would succeed and the app would not have it (or if it already had it it would keep what it had). I can see for initial restores how this could be an issue though. I'll change it to a blocking lock.